articles/mvc-mvp-mvvm/index.md

```diff
@@ -0,0 +1,52 @@
+### What are Architectural Patterns in Android?
+
+### Introduction
+
+We use various mobile and web applications in our day-to-day life. These applications are aimed to make our computational experience simple and seamless. Some of these applications are compatible with mobile operating systems such as Android, iOS and some are compatible with desktop operating systems such as Windows, macOS, Linux, etc and some of these even may be cross-platform (applications that can be operated on more than one operating system).
+
+Even though these applications may differ in aspects such as what they do, how they work, how it is built, etc.., one thing that remains the same -- that is their **unique architecture and User Interface**. These elements can serve as a key component to the overall application creation process. Having a better and unique UI can make a huge difference in how great an application is. The UI is solely not based on the developers' imagination, but also on the consumer who may be using it. The interface must be user-friendly so that it will be easier for the user to grasp the visual changes taking place in the application while using it.
+
+### Importance of Architectural Patterns
+
+A typical native android application (applications that are only compatible with Android OS) will run only on mobile devices. Now compared to desktops and laptops, mobile phones are smaller in size. So not all the content of the android app can be viewed on a single small screen. Different parts of the app are equipped in different [fragments](https://developer.android.com/guide/fragments). The app may need certain specific commands which will access these parts when the user wishes to access them. When they are accessed the whole app has to update its interface with the new data. Hence, android architecture patterns are implemented to modify these crucial parts of the code. This article deals with such architectural patterns which can be implemented to improve the overall project design.
+
+There are many benefits for an application by adding design patterns. It makes the whole source code cleaner and easier to maintain and manage. It simplifies complex codes since many functions will be nested in different classes. It also helps in covering all unit tests. This article will briefly discuss the three most popular patterns: MVC, MVP, and MVVM.

```

Comment:
```suggestion
There are many benefits for an application by adding design patterns. It makes the whole source code cleaner and easier to maintain and manage. It simplifies complex code since many functions will be nested in different classes. It also helps in covering all unit tests. This article will briefly discuss the three most popular patterns: MVC, MVP, and MVVM.
```